File: pos-system backend/src/main/java/com/ktsr/service/AuthService.java
package com.ktsr.service;

import com.ktsr.entity.User;
import com.ktsr.exceptions.UserException;
import com.ktsr.payload.DTO.UserDto;
import com.ktsr.payload.response.AuthResponse;

public interface AuthService {

    AuthResponse signup(UserDto userDto) throws UserException;

    AuthResponse login(UserDto userDto) throws UserException;

}

File: pos-system frontend/src/services/authService.js
import apiClient from "../api/httpClient.js";

export const authService = {
  login: async (payload) => {
    const response = await apiClient.post("/auth/login", payload);
    return response.data;
  },

  signup: async (payload) => {
    const response = await apiClient.post("/auth/signup", payload);
    return response.data;
  },

  profile: async () => {
    const response = await apiClient.get("/api/user/profile");
    return response.data;
  },
};
